entity_api/src/note.rs

```diff
@@ -79,7 +104,27 @@ pub async fn find_by(
         }
     }
 
-    Ok(query.all(db).await?)
+    match query.all(db).await {
+        Ok(notes) => {
+            debug!("Note found: {:?}", notes);
+
+            if !notes.is_empty() {
+                Ok(notes)
+            } else {
+                Err(Error {
+                    inner: None,
+                    error_code: EntityApiErrorCode::RecordNotFound,
+                })
+            }
+        }
+        Err(err) => {
```

Comment:
Similarly to above. when an `Err()` occurs from `query.all(db).await`, it should be propagated directly to the caller to be converted to a top level error.